### ISSUE TITLE
Wallet Backup and Restore

### ISSUE THREAD

**Issue by cryptoquick** [casey/ord#1081]
`ord wallet` is actually really useful as a user-friendly alternative to bitcoin-cli, which can be obtuse, unforgiving, and unhelpful. For example, I'm having trouble exporting my bitcoin-cli wallet created through the tutorial. I think it's an HD wallet now, so `dumpwallet` won't work, but bitcoin-cli doesn't list its own help commands, and `backupwallet` fails and provides no reason why.

So, anyway, it might be nice for lazy users like me who like to just type `ord help` and `ord wallet help`, then I could type `ord wallet backup ordinals-signet-wallet.txt` and it saves either the mnemonic or something that can be used to restore the wallet.

It'd then be nice to have a `restore` command, and also a `--dry-run` option on the restore command so I can make sure it's all backed up. Or, maybe a better UX is if importing the same file to a wallet that's already loaded, it prints the message "wallet already exists. nothing changed.", letting the user know it's reliably backed up.

**Comment by casey** [comment]
Great minds think alike! I actually just opened an issue suggesting that we should be able to backup and restore from a seed phrase, see #1063.

**Comment by cryptoquick** [comment]
Nice! Your issue actually doesn't cover backup of existing wallets, however. Feel free to mix and match this issue with that issue, I suppose. Not sure what we should do with this issue.

**Comment by casey** [comment]
Backup of existing wallets is challenging in general, I think. If we create a wallet, we can guarantee it's an HD wallet with a single root secret, which we can back up with a seed phrase. If it's an arbitrary bitcoin core wallet, then it could have multiple secrets, descriptors, etc.

I *think* that backing up existing wallets in a general way isn't possible easily, so closing in favor of #1063.

**Comment by casey** [comment]
By the way, I really like the idea of making ord a good simple wrapper wallet for Bitcoin Core. Seems like there's a real need for that.

**Comment by cryptoquick** [comment]
Well, if the xprv descriptor string is backed up to a txt file, that'd actually be an adequate substitute, imo. Maybe that could be the default behavior, and a mnemonic seed phrase could be optional.

**Comment by cryptoquick** [comment]
I finally figured out how to export descriptor xprvs:

`bitcoin-cli -signet -rpcwallet=ord listdescriptors true`

Passing true toggles between xprvs and xpubs. I discovered this here, I'm not sure where else I'd find this documented:
https://github.com/bitcoin/bitcoin/pull/21500

I'd imagine if I just > this file to json, I could then import it with importdescriptors. I think that's basically what I had in mind.
